pkg/generate/app/dockerimagelookup.go

```diff
@@ -94,6 +95,10 @@ func (r DockerClientSearcher) Search(terms ...string) (ComponentMatches, error)
 		}
 		sort.Sort(termMatches)
 
+		if r.Client == nil {
+			continue
```

Comment:
this doesn't look right...  r.Client was already used on line 76 (assuming no exact matches in the registry)